<!DOCTYPE html>
<html lang="en">
<head>
    <script async src="https://www.googletagmanager.com/gtag/js?id=G-LNJFM70WFM"></script>
    <script>
        window.dataLayer = window.dataLayer || [];
        function gtag(){dataLayer.push(arguments);}
        gtag('js', new Date());

        gtag('config', 'G-LNJFM70WFM');
    </script>

    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <meta name="description" content="Richard Lettich's personal homepage">
    <title>Richard Lettich</title>
    <style>
        body {
            font-family: Arial, sans-serif;
            margin: 0;
            padding: 0;
            display: flex;
            justify-content: center;
            align-items: flex-start;
            height: 100vh;
            background-color: #f9f9f9;
            color: #333;
        }
        .container {
            display: flex;
            width: 90%;
            max-width: 900px;
            margin-top: 40px;
            padding: 10px;
        }
        .name {
            font-family: "Courier New", Courier, monospace;
            font-size: 2em;
            margin-right: 20px;
            flex-shrink: 0;
            text-align: left;
            line-height: 1;
        }
        .content {
            max-width: 600px;
        }
        a {
            color: #007acc;
            text-decoration: none;
            transition: color 0.3s;
        }
        a:hover {
            color: #005f99;
            text-decoration: underline;
        }
        .section {
            margin-bottom: 40px;
        }
        h1, h2 {
            margin: 0 0 10px;
        }
        p {
            margin: 0 0 20px;
            line-height: 1.6;
        }
        .publications p {
            margin: 0 0 5px;
            line-height: 1.4;
        }
        @media (max-width: 600px) {
            .container {
                flex-direction: column;
                align-items: center;
            }
            .name {
                text-align: center;
                margin-right: 0;
                margin-bottom: 20px;
            }
            .content {
                text-align: center;
            }
        }
    </style>
</head>
<body>
    <div class="container">
        <div class="name">
            Richard<br>Lettich
        </div>
        <div class="content">

            <div class="section">
                <h2>About Me</h2>
                <p>I'm a CS PhD Student at UIUC.</p>
                <p>Previously I was  a software engineer on the Exabiome Project at LBNL.</p>
                <p>I did undergrad at UC Berkeley, where I received a Bachelors in Computer Science.</p>
                <p>You can contact me at <a href="mailto:rlettich@lbl.gov">rlettich@lbl.gov</a>.</p>
            </div>

            <div class="section publications">
                <h2>Interests</h2>
                <p>My research interests include machine learning, computational geometry, computational biology, numerical analysis, and high performance computing.</p>
            </div>

            <div class="section">
                <h2>Research</h2>
                <p><strong>GenomeFace: a deep learning-based metagenome binner trained on 43,000 microbial genomes</strong> </p>
                <p>Richard Lettich, Robert Egan, Robert Riley, Zhong Wang, Andrew Tritt, Leonid Oliker, Katherine Yelick, Aydın Buluç</p>
                <p><em>bioRxiv</em>, 2024. DOI: <a href="https://doi.org/10.1101/2024.02.07.579326" target="_blank" rel="noopener noreferrer">10.1101/2024.02.07.579326</a>&nbsp;&nbsp;&nbsp;<a href="https://www.biorxiv.org/content/early/2024/02/08/2024.02.07.579326.full.pdf" target="_blank" rel="noopener noreferrer">Full PDF</a></p>
                
                <p><strong>A Surprisingly Simple Method for Distributed Euclidean-Minimum Spanning Tree / Single Linkage Dendrogram Construction from High Dimensional Embeddings via Distance Decomposition</strong></p>
                <p>Richard Lettich</p>
                <p><em>arXiv</em>, 2024. DOI: <a href="https://doi.org/10.48550/arXiv.2406.01739" target="_blank" rel="noopener noreferrer">10.48550/arXiv.2406.01739</a>&nbsp;&nbsp;&nbsp;<a href="https://arxiv.org/abs/2406.01739" target="_blank" rel="noopener noreferrer">Full PDF</a></p>
                
                <p><strong>GALATIC: GPU Accelerated Sparse Matrix Multiplication over Arbitrary Semirings (GALATIC) v1.0</strong></p>
                <p>Richard Lettich</p>
                <p><em>Computer Software</em>, 2021. DOI: <a href="https://doi.org/10.11578/dc.20210902.3" target="_blank" rel="noopener noreferrer">10.11578/dc.20210902.3</a>&nbsp;&nbsp;&nbsp;<a href="https://doi.org/10.11578/dc.20210902.3" target="_blank" rel="noopener noreferrer">Access GALATIC Software</a></p>
            </div>

            <div class="section">
                <h2>GenomeFace Paper Data</h2>
                <h3>Install Guide</h3>
                <p>To use GenomeFace yourself, follow the step-by-step instructions in the <a href="gf_instructions.html" target="_blank" rel="noopener noreferrer">install guide</a>.</p>
                <h3>Interactive Taxonomic Breakdown of Multiterabase Assemblies</h3>
                <p>These interactive Sankey Diagrams showcase phylogenetic breakdowns of genome bins produced by GenomeFace on the real multiterabase assemblies featured in its paper. Each diagram displays the taxonomy of bacteria from phylum to genus level for genomes with a quality score of 50 or higher, as determined by CheckM2, with taxonomic assignments inferred by GTDB-Tk.</p>
                <p>Labels that are blank (e.g., g__ at genus level or f__ at family level) indicate the genome does not fit any known taxonomic assignment at that level.</p>
                <p><b>Interactive Sankey Diagrams:</b></p>
                <ul>
                    <li><a href="twitch_wetlands.html" target="_blank" rel="noopener noreferrer">Twitchell Island Wetlands</a></li>
                    <li><a href="harvard.html" target="_blank" rel="noopener noreferrer">Harvard Forest</a></li>
                    <li><a href="gre.html" target="_blank" rel="noopener noreferrer">Great Redox Experiment</a></li>
                </ul>
                <h3>GTDB-Tk and CheckM2 Analysis Raw Outputs</h3>
                <p>Download the <a href="genomeface_realdatasets_analysis.zip" target="_blank" rel="noopener noreferrer">zip file</a> containing the raw GTDB-Tk and CheckM2 analysis outputs for all genome binners used in the paper. GTDB-Tk provides taxonomic assignments from phylum to species level, while CheckM2 produces quality assessments of the genome bins (completeness and contamination scores). All claims and results presented in the paper for the real datasets are based on these outputs.</p>
                <h3>GenomeFace Bins</h3>
                <p>These directories contain the FASTA files of genome bins produced by GenomeFace. All genomes may be downloaded using the following command:
                <pre><code>wget -r -np -nH --cut-dirs=1 -R "index.html*" [URL]</code></pre>
                </p>
                <p><b>Download Genomes:</b></p>
                <ul>
                    <li><a href="https://portal.nersc.gov/project/m1982/twitchell_wetlands/" target="_blank" rel="noopener noreferrer">Twitchell Wetlands</a></li>
                    <li><a href="https://portal.nersc.gov/project/m1982/harvard_forest/" target="_blank" rel="noopener noreferrer">Harvard Forest</a></li>
                    <li><a href="https://portal.nersc.gov/project/m1982/GRE/" target="_blank" rel="noopener noreferrer">Great Redox Experiment</a></li>
                </ul>
            </div>
        </div>
    </div>
</body>
</html>

